cfme/common/vm_views.py

```diff
@@ -382,6 +387,14 @@ class form(BasicProvisionFormView):  # noqa
         submit_button = Button('Submit')  # Submit for 2nd page, tabular form
         cancel_button = Button('Cancel')
 
+        def _select_template(self, template_name, provider_name):
+            try:
+                row = self.parent.image_table.row(name=template_name, provider=provider_name)
+            except IndexError:
+                raise TemplateNotFound('Cannot find template "{}" for provider "{}"'
+                                       .format(template_name, provider_name))
+            row.click()
```

Comment:
**REQUIRED:** How can we click if we hit an exception finding it?

Reply:
done